packages/flutter/lib/src/gestures/multidrag.dart

```diff
@@ -328,6 +339,18 @@ abstract class MultiDragGestureRecognizer extends GestureRecognizer {
     _pointers!.remove(pointer)!.dispose();
   }
 
+  /// Checks whether or not a pointer is allowed to be tracked by this recognizer.
+  @override
+  @protected
+  bool isPointerAllowed(PointerDownEvent event) {
+    // Check for mouse button, then device kind.
+    if (inputPointerBehavior != null) {
```

Comment:
First of all, good job addressing the feedback from this PR, it has evolved a lot since the first draft.

For the line above, I'm wondering why we aren't assigning `(event.buttons & kPrimaryButton != 0)` as the default value for `inputPointerBehavior`.

If `inputPointerBehavior` defaults to `(event.buttons & kPrimaryButton != 0)`, the code below can be simplified:
```dart
  /// Checks whether or not a pointer is allowed to be tracked by this recognizer.
  @override
  @protected
  bool isPointerAllowed(PointerDownEvent event) {
    return inputPointerBehavior!(event.buttons) && super.isPointerAllowed(event); 
  }
```

Reply:
Why was this closed? I had the same thought.

Reply:
You can't have a const function constructor, so the only option is: `inputBehavior?.call(event.buttons) ?? (event.buttons & kPrimaryButton != 0)`

Reply:
You could have the default behavior as a private function in the file. Like above all the recognizers:

```dart
bool _defaultButtonAcceptBehavior(int buttons) => buttons & kPrimaryButton != 0;
```

And this could be used in the constructor as the default value.

Reply:
Can it?
<img width="1073" alt="image" src="https://user-images.githubusercontent.com/351125/196819979-a7921e80-ed69-4adf-bac9-6e2bda2c8ec0.png">

Reply:
OHHH, GOT IT!
TIL something new!

Reply:
That's awesome! Thanks!

Reply:
For monodrag (future PR), do you think I should make this function public, or copy and paste the same thing in `monodrag.dart`?

Reply:
I think it's better to copy and paste the same thing, to avoid polluting the global namespace for anyone who imports `package:flutter/gestures.dart`.